src/main/java/org/openpnp/vision/FluentCv.java

```diff
@@ -668,6 +668,8 @@ private FluentCv store(Mat mat, String... tag) {
 		if (tag != null && tag.length > 0) {
 			// Clone so that future writes to the pipeline Mat
 			// don't overwrite our stored one.
+			mat = stored.get(tag[0]);
+                        if(mat!=null) { mat.release(); }
```

Comment:
The mat continues to be used in further pipeline calls. Won't this cause it to be released early?